### PR TITLE
[EXPERIMENT] DNX Support

### PR DESCRIPTION
**Pull request by shiftkey**
This PR adds experimental support for targeting .NET Core. There's also some extra niceness in here about being able to target 

- [ ] can build on AppVeyor
- [ ] add .NET Core support to Octokit package
- [ ] add .NET Core support to Octokit.Reactive package
- [ ] target platforms in `project.json` to reduce need for managing multiple projects (and the various project hacks)
- [ ] what about executing tests locally? in CI?

### CODE REVIEW

File: Octokit/Models/Response/Committer.cs [hunk 1]
```diff
@@ -1,61 +0,0 @@
-﻿using System;
-using System.Diagnostics;
-using System.Globalization;
-
```
Comment (Haacked) [inline Octokit/Models/Response/Committer.cs]: Why was this file removed?

Comment (shiftkey) [inline Octokit/Models/Response/Committer.cs]: Duplicated file (it's also under `Models/Common`) - the new `project.json` file found this twice and it triggered a compilation error.

**Comment by shiftkey** [comment]
cc @khellang

**Comment by alfhenrik** [comment]
Oooh, this looks interesting :)

**Comment by naveensrinivasan** [comment]
Switching to new ‘dotnet’ target framework monikers   

https://github.com/aspnet/Announcements/issues/98

**Comment by shiftkey** [comment]
@naveensrinivasan yeah, that's on my list once I get to the bottom of the various build issues I've found switching over to use `project.json` for building multiple projects

**Comment by khellang** [comment]
@naveensrinivasan @shiftkey There's no point in trying to do that, cause there's no (stable) tooling that supports it. It's also worth noting that the `dotnet` TFMs are gone and replaced by `netstandard1.0` - `netstandard1.4`.  https://github.com/dotnet/corefx/blob/9c7814c301157de43e5c35d630a0352d77efde29/Documentation/project-docs/standard-platform.md

**Comment by naveensrinivasan** [comment]
Included the travis ci for the coreclr    https://github.com/octokit/octokit.net/pull/961

**Comment by khellang** [comment]
You probably want to track https://github.com/dotnet/corefx/issues/4367 for this.

**Comment by naveensrinivasan** [comment]
@shiftkey  Do you want to continue with this branch? I would like to contribute to move to coreclr.

**Comment by shiftkey** [comment]
@naveensrinivasan nah, I'm going to close this out due to frustration (and things moving around again).
